smdebug/pytorch/hook.py

```diff
@@ -54,6 +57,14 @@ def __init__(
         self.worker = self._get_worker_name()
         set_hook(self)
 
+        self.forward_start = 0
+        self.backward_start = 0
+        self.buffer = {}
+        self.profiler_enabled = profiler_enabled
+        self.profiler_steps = profiler_steps
+        self.autograd_profiler_enabled = False
+        timeline_writer = self._maybe_get_timeline_writer() 
```

Comment:
there's a _write_trace_events_summary() in core/hook.py. thought of wrapping that in a a more formal API like record_trace_event(). so all hooks could do a self.record_trace_event(). what do you think?

Reply:
changed the API name @NRauschmayr . Please rebase your PR